backend/server/public_router.go

```diff
@@ -24,17 +26,29 @@ func NewPublicRouter(cfg *config.Config, persister persistence.Persister, promet
 	e.Use(middleware.RequestID())
 	e.Use(hankoMiddleware.GetLoggerMiddleware())
 
-	if cfg.Server.Public.Cors.Enabled {
-		e.Use(middleware.CORSWithConfig(middleware.CORSConfig{
-			AllowOrigins:     cfg.Server.Public.Cors.AllowOrigins,
-			AllowMethods:     cfg.Server.Public.Cors.AllowMethods,
-			AllowHeaders:     cfg.Server.Public.Cors.AllowHeaders,
-			ExposeHeaders:    cfg.Server.Public.Cors.ExposeHeaders,
-			AllowCredentials: cfg.Server.Public.Cors.AllowCredentials,
-			MaxAge:           cfg.Server.Public.Cors.MaxAge,
-		}))
+	allowedOrigins := []string{}
+	isLocalhost := false
+	for _, v := range cfg.Webauthn.RelyingParty.Origins {
+		origin, _ := url.Parse(v)
+		if strings.HasPrefix(origin.Host, "localhost") {
+			allowedOrigins = []string{"*"}
+			isLocalhost = true
+			break
+		}
+		allowedOrigins = append(allowedOrigins, v)
 	}
-	
+	e.Use(middleware.CORSWithConfig(middleware.CORSConfig{
+		AllowOrigins: allowedOrigins,
+		// only on localhost this is allowed.
+		UnsafeWildcardOriginWithAllowCredentials: isLocalhost,
+		// Needed for cross-domain
+		ExposeHeaders: []string{"X-Auth-Token"},
```

Comment:
only needed when config.Session.EnableAuthTokenHeader is true

To be investigated: 
can 
```
c.Response().Header().Set("Access-Control-Expose-Headers", "X-Auth-Token")
```
 settings in handlers be deleted?